### PR TITLE
feat: allow updating raw-capture status to approved/rejected

### PR DESCRIPTION
**Pull request by gwynndp**

### CODE REVIEW

**Comment by gwynndp** [comment]
@Kpoke Please review when you have a chance!

File: server/handlers/rawCaptureHandler.js [hunk 1]
```diff
@@ -91,6 +132,37 @@ const rawCapturePost = async (req, res) => {
...
+  log.warn('raw capture update...', req.params.raw_capture_id);
+
+  await rawCaptureIdParamSchema.validateAsync(req.params, {
+    abortEarly: false,
+  });
+
+  const rawCaptureService = new RawCaptureService();
+  const rawCapture = await rawCaptureService.getRawCaptureById(req.body.id);
+
+  if (!rawCapture?.id) {
```
Comment (Kpoke) [inline server/handlers/rawCaptureHandler.js:145]: We don't need to get the capture and if the capture already exists

File: server/repositories/RawCaptureRepository.js [hunk 2]
```diff
@@ -109,6 +219,17 @@ class RawCaptureRepository extends BaseRepository {
 
     return result[0];
   }
+
+  async update(rawCapture) {
+    const result = await this._session
+      .getDB()
+      .update('status', rawCapture.status)
```
Comment (Kpoke) [inline server/repositories/RawCaptureRepository.js:226]: Seems like you are updating only the status, I think we should make that filter through the joi schema and this update function should remain 'generic'

File: server/handlers/rawCaptureHandler.js [hunk 3]
```diff
@@ -38,6 +54,31 @@ const rawCaptureSchema = Joi.object({
   bulk_pack_file_name: Joi.string(),
 }).unknown(false);
 
+const rawCaptureUpdateSchema = Joi.object({
```
Comment (Kpoke) [inline server/handlers/rawCaptureHandler.js:57]: The update schema should reflect the exact fields that we want updateable, rather thank doing that in the repository

File: server/handlers/rawCaptureHandler.js [hunk 4]
```diff
@@ -8,8 +8,6 @@ const {
 } = require('../utils/helper');
 
 const rawCaptureGetQuerySchema = Joi.object({
-  offset: Joi.number().integer().greater(-1),
-  limit: Joi.number().integer().greater(0),
```
Comment (Kpoke) [inline server/handlers/rawCaptureHandler.js:12]: Is there a particular reason why limit and offset were removed?

Comment (gwynndp) [reply server/handlers/rawCaptureHandler.js:12]: My bad.  I might've mistook which project I had open.

**Comment by Kpoke** [comment]
I believe we can close this PR. The endpoint to reject a raw capture currently exists while approval of raw captures is done via rabbitmq

**Pull request closed without merging**